common/djangoapps/third_party_auth/models.py

```diff
@@ -718,6 +731,25 @@ class Meta:
         verbose_name = "Provider Configuration (SAML IdP)"
         verbose_name_plural = "Provider Configuration (SAML IdPs)"
 
+    # validator method on row create
+    # looks at entity ID and slug
+    # check if any other records exist with that entity ID under a different slug
+    # prevents creation of new entry if other records found
+
+    def save(self, *args, **kwargs):
+        # Disallowing any new entries that have the same entity ID as an existing provider config
+        # Unless the slug matches.
+        # This both allows for the old architecture to create new rows on save but also prevents enterprise users from
+        # creating new configs that share entity ID's with other enterprises
+        # One consequence of this is that once a provider configuration is created, the slug is essentially locked in
+        # and unchangeable. But I blame that on bad old architecture.
+        existing_provider_configs = SAMLProviderConfig.objects.filter(
+            entity_id=self.entity_id,
+        ).exclude(slug=self.slug)
+        if existing_provider_configs:
+            raise IntegrityError(_('Entity ID already in use'))
```

Comment:
It might be a good idea to include the actual entity_id value in this error message, as well as which IdP config is already using it.